### PR TITLE
Starting GitHub Actions

### PR DESCRIPTION
**Pull request by knute-sifive**
First commit of using GitHub actions

Signed-off-by: Knute Lingaard <knute.lingaard@sifive.com>

### CODE REVIEW

**Comment by klingaard** [comment]
Hey Arup, this adds GH actions with a full ubuntu build of release/debug.  Can you approve?  I'll add MacOS after this.

**Comment by knute-sifive** [comment]
closes #6

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,7 @@ include (${SPARTA_BASE}/cmake/sparta-config.cmake)
 # Set up STF library
 set (STF_LIB_BASE ${PROJECT_SOURCE_DIR}/stf_lib)
 set (DISABLE_STF_DOXYGEN ON)
-include(${STF_LIB_BASE}/cmake/stf-config.cmake)
+#include(${STF_LIB_BASE}/cmake/stf-config.cmake)
```
Comment (arupc-vmicro) [inline CMakeLists.txt:23]: Are we not building stf_lib as a part of this flow?

**Review by arupc-vmicro — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (knute-sifive) [reply CMakeLists.txt:23]: We still are -- it's included below.  We just don't need STF-specific library definitions right now (they overlap with Sparta).

Comment (arupc) [reply CMakeLists.txt:23]: got it. thank you.

**Pull request merged**